### PR TITLE
Implement the experimental drop search after and nb search per core

### PR DESCRIPTION
**Pull request by irevoire**
# Pull Request

## Related issue
Fixes https://github.com/meilisearch/meilisearch/issues/4997
Fixes https://github.com/meilisearch/meilisearch/issues/4995

## What does this PR do?
- Add an experimental parameter to decide how many searches can be processed per core
- Add an experimental parameter to configure after how many seconds a search should be dropped

### CODE REVIEW

File: meilisearch/src/option.rs [hunk 1]
```diff
@@ -361,6 +363,22 @@ pub struct Opt {
...
+    #[clap(long, env = MEILI_EXPERIMENTAL_DROP_SEARCH_AFTER, default_value_t = default_drop_search_after())]
+    #[serde(default = "default_drop_search_after")]
+    pub experimental_drop_search_after: NonZeroUsize,
+
+    /// Experimental number of searches per core. For more information, see: <https://github.com/orgs/meilisearch/discussions/784>
+    ///
+    /// Lets you customize how many search requests can run on each core.
+    /// The default value is 4.
+    #[clap(long, env = MEILI_EXPERIMENTAL_NB_SEARCHES_PER_CORE, default_value_t = default_nb_searches_per_core())]
+    #[serde(default = "default_drop_search_after")]
```
Comment (dureuill) [inline meilisearch/src/option.rs:379]:
```suggestion
    #[serde(default = "default_nb_searches_per_core")]
```

File: meilisearch/src/option.rs [hunk 2]
```diff
@@ -361,6 +363,22 @@ pub struct Opt {
...
+    ///
+    /// Let you customize after how many seconds Meilisearch should consider a search as irrelevant and drop it.
+    /// The default value is 60.
+    #[clap(long, env = MEILI_EXPERIMENTAL_DROP_SEARCH_AFTER, default_value_t = default_drop_search_after())]
+    #[serde(default = "default_drop_search_after")]
+    pub experimental_drop_search_after: NonZeroUsize,
+
+    /// Experimental number of searches per core. For more information, see: <https://github.com/orgs/meilisearch/discussions/784>
+    ///
+    /// Lets you customize how many search requests can run on each core.
```
Comment (dureuill) [inline meilisearch/src/option.rs:376]:
```suggestion
    /// Lets you customize how many search requests can run on each core concurrently.
```

File: meilisearch/src/option.rs (hunk 1 above)
Comment (irevoire) [reply meilisearch/src/option.rs:379]: Ooof good catch, we should really find a way to write tests for the CLI 🥶

File: meilisearch/src/option.rs [hunk 3]
```diff
@@ -361,6 +363,22 @@ pub struct Opt {
     #[serde(default)]
     pub experimental_search_queue_size: usize,
 
+    /// Experimental drop search after. For more information, see: <https://github.com/orgs/meilisearch/discussions/783>
+    ///
+    /// Let you customize after how many seconds Meilisearch should consider a search as irrelevant and drop it.
```
Comment (Kerollmops) [inline meilisearch/src/option.rs:368]:
```suggestion
    /// Let you customize after how many seconds Meilisearch should consider a search request irrelevant and drop it.
```

**Review by Kerollmops — CHANGES_REQUESTED** [review]

**Review by Kerollmops — APPROVED** [review]
bors merge

**Pull request merged**